pkg/serving/config_changes.go

```diff
@@ -233,6 +233,17 @@ func UpdateLabels(service *servingv1alpha1.Service, template *servingv1alpha1.Re
 	return nil
 }
 
+// UpdateServiceAccountName updates the service account name used for the corresponding knative service
+func UpdateServiceAccountName(template *servingv1alpha1.RevisionTemplateSpec, serviceAccountName string) error {
+	serviceAccountName = strings.TrimSpace(serviceAccountName)
+	if serviceAccountName == "-" {
```

Comment:
Related to my comment above in description. Empty string for unsettling would make this read better

Reply:
I've changed the default value to an empty string.